### DISCUSSION TITLE
Rest client and server null handling

### DISCUSSION THREAD

**Discussion by robp94** [quarkusio/quarkus#25369] [Q&A] [answered]
Resteays classic converts a null return automatically to a 404.

Is there an Option for the client to do the same in reverse, getting a 404 and returning null.

[a comment by an automated account is not shown]

**Reply by michalszynkiewicz** [reply]
You can use `ResponseExceptionMapper` to achieve this, take a look at https://quarkus.io/guides/rest-client-reactive#exception-handling

**Reply by robp94** [reply]
This does not work for me, or I am doing something wrong.

If I understood you correctly, you meant something like this.

```
public class ExceptionHandler implements ResponseExceptionMapper<RuntimeException> {
    @Override
    public RuntimeException toThrowable(Response response) {
        return null;
    }

    @Override
    public boolean handles(int status, MultivaluedMap<String, Object> headers) {
        return status==Response.Status.NOT_FOUND.getStatusCode();
    }
}
```
I registered the mapper with @RegisterProvider, and it gets called.

But when I call the client, I still get an exception like `Received: 'Not Found, status code 404' when invoking: Rest Client method` instead of null.

**Reply by michalszynkiewicz** [reply]
oh, you're right, sorry.

So I guess you're left with ClientResponseFilter. You should be able to check the response status and then modify the response.
I haven't tried it though.

**Reply by robp94** [reply]
Deactivating the default exception mapper seems to work `microprofile.rest.client.disable.default.mapper=true`
